src/components/Popup.js

```diff
@@ -32,6 +33,14 @@ export default class Popup {
     }
   };
 
+  renderLoading() {
```

Comment:
Needs correction: the common popup should not have methods for rendering loading, since it renders any popup, not only popups with submit buttons. Please either add a new class PopupWithSubmit or add the methods for rendering loading to confirmation popup and popup with form 

Also note, that the button texts should be different for each popup and in ` endLoading` method you should return the initial button text. You can save the initial text in the constructor and use it:

```js
constructor() {
  ..
  this._submitButton = this._button.textContent;
```